tcc/src/main.c

```diff
@@ -67,38 +67,44 @@ main_end(void)
 	kill_stray();
 }
 
+typedef struct _t_env_pair {
+	const char *name;
+	const char *value;
+} t_env_pair;
 
-/*
- * MAIN INITIALISATION ROUTINE
- *
- * This is the initialisation routine called at the very start of the program.
- */
 
-static void
-main_start(char *prog)
+void init_prefix_envvars(const char *root_path)
 {
-	size_t i;
-
-	struct {
-		const char *name;
-		const char *value;
-	} a[] = {
-		{ "PREFIX",         PREFIX         },
-		{ "PREFIX_BIN",     PREFIX_BIN     },
-		{ "PREFIX_LIB",     PREFIX_LIB     },
-		{ "PREFIX_LIBEXEC", PREFIX_LIBEXEC },
-		{ "PREFIX_SHARE",   PREFIX_SHARE   },
-		{ "PREFIX_INCLUDE", PREFIX_INCLUDE },
-		{ "PREFIX_MAN",     PREFIX_MAN     },
-		{ "PREFIX_TSPEC",   PREFIX_TSPEC   },
-		{ "PREFIX_STARTUP", PREFIX_STARTUP },
-		{ "PREFIX_ENV",     PREFIX_ENV     },
-		{ "PREFIX_API",     PREFIX_API     },
-		{ "PREFIX_LPI",     PREFIX_LPI     },
-		{ "PREFIX_SYS",     PREFIX_SYS     },
-		{ "PREFIX_TMP",     PREFIX_TMP     },
-		{ "PREFIX_MAP",     PREFIX_MAP     },
+	const t_env_pair a[] = {
+		{ "PREFIX_BIN",     path_join(root_path, "bin")            },
+		{ "PREFIX_LIB",     path_join(root_path, "lib")            },
+		{ "PREFIX_LIBEXEC", path_join(root_path, "libexec")        },
+		{ "PREFIX_SHARE",   path_join(root_path, "share")          },
+		{ "PREFIX_INCLUDE", path_join(root_path, "include")        },
+		{ "PREFIX_MAN",     path_join(root_path, "man")            },
+		{ "PREFIX_TSPEC",   path_join(root_path, "share/tspec")    },
+		{ "PREFIX_STARTUP", path_join(root_path, "lib/tcc/startup")},
+		{ "PREFIX_ENV",     path_join(root_path, "lib/tcc/env")    },
+		{ "PREFIX_API",     path_join(root_path, "lib/tcc/api")    },
+		{ "PREFIX_LPI",     path_join(root_path, "lib/tcc/lpi")    },
+		{ "PREFIX_SYS",     path_join(root_path, "lib/tcc/sys")    },
+		{ "PREFIX_MAP",     path_join(root_path, "lib/tcc/map")    },
+	};
```

Comment:
If we're hardcoding the strings for relative paths here, that means we no longer use the `PREFIX_` defines. So I think all the `-DPREFIX_*=...` arguments can go from src/Makefile. And that means their corresponding Makefile variables can go, too. And then they can go from the top-level Makefile, too.

Reply:
Ok, will look at this tomorrow

Reply:
I removed the -DPREFIX_ defines and some make arguments from the top-level Makefile. But I don't think I can remove much from the .mk files as they are used as target folder to install into. (I tried removing some but a lot of them are used)

Reply:
currently rewriting some things so that tendra can compile the code as well